### PR TITLE
SQL: Migrate (MS/My/Postgres)SQL configuration pages from Angular to React

### PR DESCRIPTION
**Pull request by oscarkilhed**
This is a migration to react of the configuration pages for the core SQL datasources (mysql, mssql, postgres). Most things should be fairly similar in look and feel to the old configuration pages. But some elements have changed in size because we're not using our standard react components. The checkboxes has also been swapped to switches for all cases.

How the settings for the datasource are structured has been kept intact to avoid having to migrate the datasource instances. There are however some room for improvements here in terms of structure and naming. I believe it's best to make these changes at a later time so that any issues that arises with this migration are easier to identify.

Part of #51476

### CODE REVIEW

File: .betterer.results [hunk 1]
```diff
@@ -8061,15 +8056,6 @@ exports[`better eslint`] = {
...
     ],
-    "public/app/plugins/datasource/postgres/config_ctrl.ts:5381": [
-      [0, 0, 0, "Unexpected any. Specify a different type.", "0"],
-      [0, 0, 0, "Unexpected any. Specify a different type.", "1"],
-      [0, 0, 0, "Unexpected any. Specify a different type.", "2"],
-      [0, 0, 0, "Unexpected any. Specify a different type.", "3"],
-      [0, 0, 0, "Unexpected any. Specify a different type.", "4"],
-      [0, 0, 0, "Unexpected any. Specify a different type.", "5"],
-      [0, 0, 0, "Unexpected any. Specify a different type.", "6"]
-    ],
```
Comment (joshhunt) [inline .betterer.results:8072]: 📉  NOICE

File: public/app/plugins/datasource/postgres/datasource.ts [hunk 2]
```diff
@@ -185,11 +185,23 @@ export class PostgresDatasource extends DataSourceWithBackend<PostgresQuery, Pos
   }
 
   getVersion(): Promise<any> {
-    return lastValueFrom(this._metaRequest("SELECT current_setting('server_version_num')::int/100"));
+    return lastValueFrom(this._metaRequest("SELECT current_setting('server_version_num')::int/100")).then((value) => {
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/postgres/datasource.ts:188]: So I was looking at these changes and wondered why are these needed. So we had a code that checked for version in the config page code that was not working. What do you all think, do we need that? @oscarkilhed @scottlepp

Comment (scottlepp) [reply public/app/plugins/datasource/postgres/datasource.ts:188]: looks like that is/was used for feature detection/validation in postgres.  i'm a bit reluctant to remove it since it was already there, but not sure if it is absolutely necessary

Comment (zoltanbedi) [reply public/app/plugins/datasource/postgres/datasource.ts]: I added back that code that found the version and set timescale db to true if it was timescale.

**Review by scottlepp — APPROVED** [review]

**Review by zoltanbedi — APPROVED** [review]
